src/aggregation/sum.c

```diff
@@ -0,0 +1,267 @@
+#include "postgres.h"
+
+#include <math.h>
+
+#include "catalog/pg_type.h"
+
+#include "pg_diffix/aggregation/common.h"
+#include "pg_diffix/aggregation/summable.h"
+#include "pg_diffix/config.h"
+#include "pg_diffix/query/anonymization.h"
+
+static double finalize_sum_result(const SummableResultAccumulator *accumulator)
+{
+  return accumulator->sum_for_flattening + accumulator->noise_with_max_sd;
+}
+
+/*-------------------------------------------------------------------------
+ * Aggregation callbacks
+ *-------------------------------------------------------------------------
+ */
+
+typedef struct SumState
+{
+  AnonAggState base;
+  int trackers_count;
+  Oid summand_type;
+  ContributionTrackerState *trackers[FLEXIBLE_ARRAY_MEMBER];
+} SumState;
+
+static void sum_final_type(Oid primary_arg_type, Oid *type, int32 *typmod, Oid *collid)
+{
+  switch (primary_arg_type)
+  {
+  case INT2OID:
+  case INT4OID:
+    *type = INT8OID;
+    break;
+  case INT8OID:
+    // FIXME should  be numeric
+    *type = INT8OID;
+    break;
+  case NUMERICOID:
+    FAILWITH("sum(numeric) not supported");
+    break;
+  case FLOAT4OID:
+    *type = FLOAT4OID;
+    break;
+  case FLOAT8OID:
+    *type = FLOAT8OID;
+    break;
+  default:
+    Assert(false);
+    *type = FLOAT8OID;
+  }
+  *typmod = -1;
+  *collid = 0;
+}
+
+static const int SUM_VALUE_INDEX = 1;
+static const int SUM_AIDS_OFFSET = 2;
+
+static AnonAggState *sum_create_state(MemoryContext memory_context, ArgsDescriptor *args_desc)
+{
+  MemoryContext old_context = MemoryContextSwitchTo(memory_context);
+
+  int trackers_count = args_desc->num_args - SUM_AIDS_OFFSET;
+  SumState *state = palloc0(sizeof(SumState) + trackers_count * sizeof(ContributionTrackerState *));
+  state->trackers_count = trackers_count;
+  state->summand_type = args_desc->args[SUM_VALUE_INDEX].type_oid;
+  ContributionDescriptor typed_sum_descriptor = {0};
+  switch (state->summand_type)
+  {
+
+  case INT2OID:
+  case INT4OID:
+  case INT8OID:
+    typed_sum_descriptor = integer_descriptor;
+    break;
+  case NUMERICOID:
+    FAILWITH("sum(numeric) not supported");
+    break;
+  case FLOAT4OID:
+  case FLOAT8OID:
+    typed_sum_descriptor = real_descriptor;
+    break;
+  default:
+    Assert(false);
+    typed_sum_descriptor = real_descriptor;
+  }
+  for (int i = 0; i < trackers_count; i++)
```

Comment:
```suggestion
  }
  
  for (int i = 0; i < trackers_count; i++)
```

I find it helps for separating large blocks.